### PR TITLE
(MODULES-10765) Implement public key certificate support

### PR DESCRIPTION
**Pull request by bastelfreak**

### CODE REVIEW

**Comment by bastelfreak** [comment]
I went ahead and implemented all types from https://raw.githubusercontent.com/openssh/openssh-portable/master/PROTOCOL.certkeys

**Comment by bastelfreak** [comment]
I tested this also in my own environment and can verify that this doesn't break anymore if such keys are present on the system.

**Comment by bastelfreak** [comment]
Hi people. I notice that there are no code owners defined here. Is this on purpose or was that missed? I thought all maintained Puppet repos now have that.

**Comment by GabrielNagy** [comment]
There is a codeowners file https://github.com/puppetlabs/puppetlabs-sshkeys_core/blob/main/CODEOWNERS so a review request should have been triggered, I'm not sure why it didn't happen.

File: spec/unit/type/ssh_authorized_key_spec.rb [hunk 1]
```diff
@@ -87,7 +87,8 @@
         :'ecdsa-sha2-nistp521',
         :ed25519, :'ssh-ed25519',
         :'ecdsa-sk', :'sk-ecdsa-sha2-nistp256@openssh.com',
-        :'ed25519-sk', :'sk-ssh-ed25519@openssh.com'
+        :'ed25519-sk', :'sk-ssh-ed25519@openssh.com',
+        :'ssh-rsa-cert-v01@openssh.com'
```
Comment (GabrielNagy) [inline spec/unit/type/ssh_authorized_key_spec.rb:91]: Can you add the rest of the key types here?

Comment (bastelfreak) [reply spec/unit/type/ssh_authorized_key_spec.rb]: done

**Review by GabrielNagy — APPROVED** [review]

**Comment by GabrielNagy** [comment]
PR looks good :+1:, thanks for the contribution!

**Comment by bastelfreak** [comment]
@GabrielNagy could you kick off a new release soonish?

**Comment by GabrielNagy** [comment]
sure @bastelfreak, as it happens we already have the release ticket in our sprint 😄 gonna jump on it

**Pull request merged**